### PR TITLE
Improve Node logic

### PR DESCRIPTION
**Pull request by Milo123459**
This removes a lot of unneeded Results, replacing them mostly with Options. It also cleans up some of the code into it's own function for it to be used by other providers.

### CODE REVIEW

File: src/providers/node.rs [hunk 1]
```diff
@@ -163,48 +133,89 @@ impl NodeProvider {
...
+        }
     }
 }
 
-fn version_number_to_pkg(version: &u32) -> Result<Option<String>> {
-    if AVAILABLE_NODE_VERSIONS.contains(version) {
-        Ok(Some(format!("nodejs-{}_x", version)))
+fn version_number_to_pkg(version: &u32) -> String {
+    if version % 2 == 1 {
+        String::from("nodejs")
```
Comment (JakeCooper) [inline src/providers/node.rs:203]: I think we should alert if we fallback from a non-stable version?

**Review by JakeCooper — APPROVED** [review]
Looks good one tiny comment!

File: src/providers/node.rs [hunk 2]
```diff
@@ -163,48 +133,89 @@ impl NodeProvider {
...
```
Comment (wyzlle) [inline src/providers/node.rs:203]:
```suggestion
        DEFAULT_NODE_PKG_NAME.to_string()
```

Comment (JakeCooper) [reply src/providers/node.rs:203]: Yea good call

Comment (Milo123459) [reply src/providers/node.rs:203]: `DEFAULT_NODE_PKG_NAME` is just `nodejs`?

File: src/providers/node.rs [hunk 3]
```diff
@@ -163,48 +134,89 @@ impl NodeProvider {
...
 }
 
-fn version_number_to_pkg(version: &u32) -> Result<Option<String>> {
+fn version_number_to_pkg(version: &u32) -> String {
     if AVAILABLE_NODE_VERSIONS.contains(version) {
-        Ok(Some(format!("nodejs-{}_x", version)))
+        format!("nodejs-{}_x", version)
     } else {
-        bail!("Node version {} is not available", version);
+        println!("No node version could be matched, defaulting to {DEFAULT_NODE_PKG_NAME}");
```
Comment (wyzlle) [inline src/providers/node.rs:205]:
```suggestion
        println!("\x1b[93mWARNING: No node version could be matched, defaulting to {DEFAULT_NODE_PKG_NAME}\x1b[0m");
```

File: src/providers/node.rs [hunk 4]
```diff
@@ -7,12 +7,12 @@ use crate::nixpacks::{
     nix::Pkg,
     phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
 };
-use anyhow::{bail, Result};
+use anyhow::Result;
 use regex::Regex;
 use serde::{Deserialize, Serialize};
 
+const DEFAULT_NODE_PKG_NAME: &'static &str = &"nodejs";
```
Comment (wyzlle) [inline src/providers/node.rs:14]:
```suggestion
const DEFAULT_NODE_PKG_NAME: &'static &str = &"nodejs-16_x";
```

Comment (Milo123459) [reply src/providers/node.rs:14]: Not needed

Comment (wyzlle) [reply src/providers/node.rs:14]: I believe this is required. If nix updates default version it will break nixpack's reproducibility.

**Pull request merged**